### PR TITLE
Web: fix git blame for files that have `/stream/` in their path

### PR DESCRIPTION
**Pull request by camdencheek**
The URL path for streaming blame is ambiguous when the file path contains `/stream/`. The pattern looks like `/blame/<repo_name>@<revision>/stream/file/path.txt`. However, if the file contains `/stream/`, the revision is matched greedily up until the last stream, so we end up with a revision that looks like `81af3g/stream/my/path`, which is in invalid revision that results in a 404. 

This makes the URL pattern unambiguous by adding a `/-/` element after the revision, which is not allowed in a revision name and acts as a path separator. So now, the pattern looks like `/blame/<repo_name>@<revision>/-/stream/file/path.txt`.

Note, this is a public-facing breaking change, but I don't think it really matters since I don't expect any customers use our streaming blame API

Fixes SRCH-738

## Test plan

Tested that blame requests do not fail for files that contain `/stream/`

## Changelog

- Fixed an issue where blame view would error when the file contains the path element `/stream/`

### CODE REVIEW

File: client/web/src/repo/blame/shared.ts [hunk 1]
```diff
@@ -147,10 +147,10 @@ function fetchBlameHunks(repoName: string, revision: string, filePath: string):
 function fetchRawBlameHunks(repoName: string, revision: string, filePath: string): Observable<RawStreamHunk[]> {
     const repoAndRevisionPath = `/${repoName}${revision ? `@${revision}` : ''}`
     return new Observable<RawStreamHunk[]>(subscriber => {
-        fetchEventSource(`/.api/blame${repoAndRevisionPath}/stream/${filePath}`, {
+        fetchEventSource(`/.api/blame${repoAndRevisionPath}/-/stream/${filePath}`, {
             method: 'GET',
             headers: {
-                'X-Requested-With': 'Sourcegraph',
+                ...window.context.xhrHeaders,
```
Comment (camdencheek) [inline client/web/src/repo/blame/shared.ts:153]: Unrelated drive-by: we're already adding `X-Requested-With`, so this should be okay, but we should prefer to use the xhr headers sent by the backend.

**Review by vovakulikov — APPROVED** [review]
It looks like that path as a URL never has a state where there is no problem with it. LGTM
